src/bin/pg_autoctl/pgctl.c

```diff
@@ -586,18 +588,32 @@ log_program_output(Program prog)
 {
 	if (prog.stdout != NULL)
 	{
-		log_info("%s", prog.stdout);
+		char *outLines[BUFSIZE];
+		int lineCount = splitLines(prog.stdout, outLines, BUFSIZE);
+		int lineNumber = 0;
+
+		for (lineNumber = 0; lineNumber < lineCount; lineNumber++)
+		{
+			log_info("%s", outLines[lineNumber]);
+		}
 	}
 
 	if (prog.stderr != NULL)
 	{
-		if (prog.returnCode == 0)
-		{
-			log_info("%s", prog.stderr);
-		}
-		else
+		char *errorLines[BUFSIZE];
```

Comment:
we have way too many repetitions of the same code block. Would it be possible to encapsulate that into function and call just that ?

Reply:
I have improved our `log_program_output` to make it usable from more places, reducing the code copying.